File: ICS_101/practice/final/numsumreverse/numsumreverse.dart
import 'dart:io';
import 'dart:math';

int sum_digits(int n) {
  if (n == 0) return 0;
  return (n % 10) + sum_digits(n ~/ 10);
}

num reverse_digits(int n) {
  if (n < 10) return n;
  return (n % 10) * pow(10, (n.toString().length - 1)) +
      reverse_digits(n ~/ 10);
}

void main(List<String> args) {
  stdout.write('Enter a number: ');
  int n = int.parse(stdin.readLineSync() ?? '0');

  stdout.writeln('Sum: ${sum_digits(n)}');
  print(n);
  stdout.writeln('Reverse: ${reverse_digits(n)}');
}

File: ICS_101/practice/final/numsumreverse/numsumreverse.c
#include <stdio.h>

int sum_digits(int n) {
  if (n == 0) return 0;
  return (n % 10) + sum_digits(n / 10);
}

int reverse_digits(int n) {
  if (n == 0) return 0;
  return (n % 10) * 10 + reverse_digits(n / 10);
}

int main(int argc, char *argv[]) {
  int n;
  printf("Enter a number : ");
  scanf("%d", &n);

  printf("Sum: %d", sum_digits(n));
  printf("Reverse: %d", reverse_digits(n));

  return 0;
}
